torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -222,6 +224,13 @@ class CompiledNodeArgs {
     }
   }
   template <typename T>
+  void collect(const std::unordered_set<T>& t) {
+    collect_size(t.size());
+    for (const T& i : t) {
```

Comment:
Is iteration order deterministic?

Reply:
not if the user overrides the default unordered_set. i'll have to sort it

Reply:
we will get recompiles for non-determinstic orders